builtin/providers/azurerm/resource_arm_loadbalancer_rule.go

```diff
@@ -0,0 +1,342 @@
+			},
+
+			"probe_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"enable_floating_ip": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"idle_timeout_in_minutes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+
+			"load_distribution": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmLoadbalancerRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	lbClient := client.loadBalancerClient
+
+	loadBalancer, exists, err := retrieveLoadbalancerById(d.Get("loadbalancer_id").(string), meta)
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer By ID {{err}}", err)
+	}
+	if !exists {
+		d.SetId("")
+		log.Printf("[INFO] LoadBalancer %q not found. Removing from state", d.Get("name").(string))
+		return nil
+	}
+
+	_, _, exists = findLoadBalancerRuleByName(loadBalancer, d.Get("name").(string))
+	if exists {
+		return fmt.Errorf("A LoadBalancer Rule with name %q already exists.", d.Get("name").(string))
+	}
+
+	newLbRule, err := expandAzureRmLoadbalancerRule(d, loadBalancer)
+	if err != nil {
+		return errwrap.Wrapf("Error Exanding LoadBalancer Rule {{err}}", err)
+	}
+
+	lbRules := append(*loadBalancer.Properties.LoadBalancingRules, *newLbRule)
+	loadBalancer.Properties.LoadBalancingRules = &lbRules
+	resGroup, loadBalancerName, err := resourceGroupAndLBNameFromId(d.Get("loadbalancer_id").(string))
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer Name and Group: {{err}}", err)
+	}
+
+	_, err = lbClient.CreateOrUpdate(resGroup, loadBalancerName, *loadBalancer, make(chan struct{}))
+	if err != nil {
+		return errwrap.Wrapf("Error Creating/Updating LoadBalancer {{err}}", err)
+	}
+
+	read, err := lbClient.Get(resGroup, loadBalancerName, "")
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer {{err}}", err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read LoadBalancer %s (resource group %s) ID", loadBalancerName, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	log.Printf("[DEBUG] Waiting for LoadBalancer (%s) to become available", loadBalancerName)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"Accepted", "Updating"},
+		Target:  []string{"Succeeded"},
+		Refresh: loadbalancerStateRefreshFunc(client, resGroup, loadBalancerName),
+		Timeout: 10 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for LoadBalancer (%s) to become available: %s", loadBalancerName, err)
+	}
+
+	return resourceArmLoadbalancerRuleRead(d, meta)
+}
+
+func resourceArmLoadbalancerRuleRead(d *schema.ResourceData, meta interface{}) error {
+	loadBalancer, exists, err := retrieveLoadbalancerById(d.Id(), meta)
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer By ID {{err}}", err)
+	}
+	if !exists {
+		d.SetId("")
```

Comment:
Is there an equivalent case where the LoadBalancer exists, but the LoadBalancer_rule does not?
I can't see another `d.SetId("")` in the resourceArmLoadbalancerRuleRead().